src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperPass.cs

```diff
@@ -15,6 +15,12 @@ public class ViewComponentTagHelperPass : IntermediateNodePassBase, IRazorOptimi
 
         protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentIntermediateNode documentNode)
         {
+            if (FileKinds.IsComponent(codeDocument.GetFileKind()))
```

Comment:
Are there other passes we can apply this heuristic on? E.g. `ModelExpressionPass`.

Reply:
Yeah, it does look like we could do more here. This one popped up when I was debugging / profiling.